homeassistant/components/sensor/magicseaweed.py

```diff
@@ -0,0 +1,258 @@
+            forecast.get('solidRating') + forecast.get('fadedRating'),
+            forecast.get('solidRating'))
+
+        return {
+            'air_pressure': "{}{}".format(condition.get('pressure'),
+                                          condition.get('unitPressure')),
+            'air_temp': "{}° {}".format(condition.get('temperature'),
+                                        condition.get('unit')),
+            'rating': rating,
+            'begins': occurs,
+            'issued': utc_issue,
+            'max_breaking_height': swell.get('maxBreakingHeight'),
+            'min_breaking_height': swell.get('minBreakingHeight'),
+            'probability': "{}%".format(swell.get('probability')),
+            'swell_period': "{} seconds".format(
+                swell.get('components', {})
+                .get('combined', {})
+                .get('period', {})),
+            'wind_chill': "{}°".format(wind.get('chill')),
+            'wind_direction': "{}° {}".format(wind.get('direction'),
+                                              wind.get('compassDirection')),
+            'wind_gusts': "{} {}".format(wind.get('gusts'), wind.get('unit')),
+            'wind_speed': "{} {}".format(wind.get('speed'), wind.get('unit')),
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION
+            }
+
+    def update(self):
+        """Get the latest data from Magicseaweed and updates the states."""
+        self.data.update()
+        if self.hour is None:
+            forecast = self.data.currently
+        else:
+            forecast = self.data.hourly[self.hour]
+
+        swell = forecast.get('swell')
+        self._unit_of_measurement = swell.get('unit', None)
+        if self.type == 'min_breaking_swell':
+            self._state = swell.get('minBreakingHeight')
+        elif self.type == 'max_breaking_swell':
+            self._state = swell.get('maxBreakingHeight')
+        elif self.type == 'swell_forecast':
+            summary = "{} - {}".format(
+                swell.get('minBreakingHeight'),
+                swell.get('maxBreakingHeight'))
+            self._state = summary
+
+
+class MagicSeaweedData(object):
+    """Get the latest data from MagicSeaweed."""
+
+    def __init__(self, api_key, spot_id, units, interval):
+        """Initialize the data object."""
+        self._api_key = api_key
+        self._spot_id = spot_id
+        self.currently = None
+        self.hourly = {}
+        self.params = {'spot_id': self._spot_id,
+                       'units': units}
+
+        # Apply throttling to methods using configured interval
+        self.update = Throttle(interval)(self._update)
+
+    def _update(self):
+        """Get the latest data from MagicSeaweed."""
+        try:
+            now = datetime.now()
+            plus_24 = now + timedelta(hours=24)
+            self.params['start'] = now.timestamp()
+            self.params['end'] = plus_24.timestamp()
+            response = requests.get(API_URL.format(self._api_key),
```

Comment:
Your integration shouldn't directly call APIs, interact with services, etc. The common practice is to extract this logic into a standalone package (with its own exceptions), upload it to PyPi, and use _that_ instead.